### PR TITLE
Api connectors

### PR DESCRIPTION
**Pull request by millboh**
New API connector supporting JSON or Text content-types.

### CODE REVIEW

File: runner/src/main/java/org/ananas/runner/legacy/steps/files/TextConnector.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
   public TextConnector(
       String stepId, String url, Pipeline pipeline, boolean doSampling, boolean isTest) {
     super(StepType.Connector);
-    final Schema schema = Schema.builder().addField("text", Schema.FieldType.STRING).build();
+    final Schema schema = Schema.builder().addStringField("text").build();
```
Comment (bhou) [inline runner/src/main/java/org/ananas/runner/legacy/steps/files/TextConnector.java]: Please move this connector from `legacy.files` to `steprunner.files` package.

File: runner/src/main/java/org/ananas/runner/legacy/core/StepConfig.java [hunk 2]
```diff
@@ -55,9 +55,12 @@
 
   // API
   public static final String API_METHOD = "method";
+  public static final String API_FORMAT = "format";
```
Comment (bhou) [inline runner/src/main/java/org/ananas/runner/legacy/core/StepConfig.java]: We should stop using the unified StepConfig class, instead, create a dedicated `APIStepConfig` class in the `steprunner` package, in order to separate concerns for each step.

File: runner/src/main/java/org/ananas/runner/misc/StepConfigHelper.java [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
 
 public class StepConfigHelper {
+
+  public static <T> T getGenericsConfig(Map<String, T> config, String name, T defaultValue) {
+    if (config.containsKey(name) && config.get(name) != null) {
+      return config.get(name);
+    }
+    return defaultValue;
+  }
+
```
Comment (bhou) [inline runner/src/main/java/org/ananas/runner/misc/StepConfigHelper.java]: Very useful!

File: runner/src/main/java/org/ananas/runner/legacy/steps/files/TextConnector.java (hunk 1 above)
Comment (millboh) [reply runner/src/main/java/org/ananas/runner/legacy/steps/files/TextConnector.java]: acceptance test are added

Comment (millboh) [reply runner/src/main/java/org/ananas/runner/legacy/steps/files/TextConnector.java]: this TextConnector is not used. I refactored it but it's still legacy.

**Pull request merged**